lib/imagemounter/index.js

```diff
@@ -0,0 +1,119 @@
+import { readFile, existsSync, lstatSync, createReadStream } from 'fs';
+import { BaseServicePlist } from '../base-service';
+import B from 'bluebird';
+import log from '../logger';
+
+const MOBILE_IMAGE_MOUNTER_SERVICE_NAME = 'com.apple.mobile.mobile_image_mounter';
+
+class ImageMounter extends BaseServicePlist {
+
+    _checkReturnError(ret) {
+        if (ret.Error) {
+            throw new Error(ret.Error);
+        }
+    }
+
+    /**
+     * Lookup for mounted images.
+     * * @param {string?} imageType Type of image, `Developer` by default.
+     * @returns {Promise<Buffer[]>} Signature of each mounted image.
+     */
+    async lookup(imageType = 'Developer') {
+        const ret = await this._plistService.sendPlistAndReceive({
+            Command: 'LookupImage',
+            ImageType: imageType
+        });
+        this._checkReturnError(ret);
+        return ret.ImageSignature || [];
+    }
+
+    /**
+     * Check if developer image is mounted.
+     */
+    async developerMounted() {
+        return (await this.lookup()).length > 0;
+    }
+
+    /**
+     * Mount image for device.
+     * @param {string} imageFilePath The file path of image.
+     * @param {string} imageSignatureFilePath The signature file path of given `DeveloperDiskImage.dmg`
+     * @param {string?} imageType Type of image, `Developer` by default.
+     */
+    async mount(imageFilePath, imageSignatureFilePath, imageType = 'Developer') {
+        //check file stats
+        if (!existsSync(imageFilePath)) {
+            throw Error(`The image file provided not exists: ${imageFilePath}`);
+        }
+        const imageFileStat = lstatSync(imageFilePath);
+        if (imageFileStat.isDirectory()) {
+            throw Error(`The given image file is expected to be a file, a directory was given: ${imageFilePath}`);
+        }
+        if (!existsSync(imageSignatureFilePath)) {
+            throw Error(`The signature file provided not exists: ${imageFilePath}`);
+        }
+        if (lstatSync(imageSignatureFilePath).isDirectory()) {
+            throw Error(`The given signature file is expected to be a file, a directory was given: ${imageFilePath}`);
+        }
+        //read signature
+        const signature = await new B((resolve, reject) => {
+            readFile(imageSignatureFilePath, (err, data) => {
+                if (err != null) {
+                    reject(err);
+                } else {
+                    resolve(data);
+                }
+            });
+        });
+        const mountedImages = await this.lookup(imageType);
+        if (mountedImages.find((mountedSignature) => signature.equals(mountedSignature))) {
```

Comment:
is `signature` a string? should we use `===` instead?

Reply:
if these two are buffers then it would probably make sense to use https://www.w3schools.com/nodejs/met_buffer_compare.asp